### DISCUSSION TITLE
Adding elements to list that is not empty

### DISCUSSION THREAD

**Discussion by siredmar** [kubevela/kubevela#3800] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

I'm always frustrated when i have a json array that should be filled but has some predefined values.

e.g. this wont work and gives the error: `failed to render <file.cue>, reason: failed to parse CUE: multiple comprehensions per list not yet supported`

```
 if parameter["volumes"] != _|_ {
    volumes: [
    {
	"name": "nats-credentials",
	"secret": {
		"secretName": context.appName+"."+context.name
	}
    },
    for v in parameter.volumes {
    {
	name: v.name
	secret: {
		defaultMode: v.defaultMode
		secretName:  v.secretName
		if v.items != _|_ {
			items: v.items
		}
	}
    }
]
							
```

**Describe the solution you'd like**

I'd like to add predefined values to a list like in the code above.

**Describe alternatives you've considered**

For simple types like string, an intermediate list can be created with the predefined values in it, and using the parameters. Then flatten it as input. I don't know if this migt work for more complex structs.

**Additional context**

**Comment by lujiajing1126** [comment, marked as answer]
As hinted by the official implementation, this can be done by using `+` operator.

https://github.com/oam-dev/kubevela/blob/master/vela-templates/definitions/internal/trait/env.cue#L44-L59

I've tested with the following code snippet,

```cue
if parameter["env"] == _|_ {
    env: predefinedEnv
}

if parameter["env"] != _|_ {
    env: predefinedEnv + parameter.env
}
```

**Reply by delprofundo** [reply]
LEGEND! :)

**Comment by wonderflow** [comment]
> As hinted by the official implementation, this can be done by using `+` operator.
> 
> https://github.com/oam-dev/kubevela/blob/master/vela-templates/definitions/internal/trait/env.cue#L44-L59
> 
> I've tested with the following code snippet,
> 
> ```cue
> if parameter["env"] == _|_ {
>     env: predefinedEnv
> }
> 
> if parameter["env"] != _|_ {
>     env: predefinedEnv + parameter.env
> }
> ```

Thanks! I think this should be the right answer. 

Closing, @siredmar feel free to re-open it if the issue still exists.
